spec/controllers/basespace_controller_spec.rb

```diff
@@ -0,0 +1,239 @@
+require 'rails_helper'
+
+RSpec.describe BasespaceController, type: :controller do
+  create_users
+
+  context "Admin user" do
+    before do
+      sign_in @admin
+    end
+
+    describe "GET oauth" do
+      before do
+        allow(HttpHelper).to receive(:post).and_return("access_token" => "12345")
```

Comment:
Note: a general good practice is to assign these strings to constants to make sure they are in sync with checks below...